sqlmesh/core/engine_adapter/mssql.py

```diff
@@ -206,3 +208,80 @@ def _truncate_table(self, table_name: TableName) -> str:
     def _to_sql(self, expression: exp.Expression, quote: bool = True, **kwargs: t.Any) -> str:
         sql = super()._to_sql(expression, quote=quote, **kwargs)
         return f"{sql};"
+
+    @overload
+    def execute_and_fetch(
+        self,
+        expression: t.Union[str, exp.Expression],
+        fetch_one: Literal[False] = ...,
+        ignore_unsupported_errors: bool = False,
+        quote_identifiers: bool = True,
+        **kwargs: t.Any,
+    ) -> t.List[t.Tuple]:
+        ...
+
+    @overload
+    def execute_and_fetch(
+        self,
+        expression: t.Union[str, exp.Expression],
+        fetch_one: Literal[True],
+        ignore_unsupported_errors: bool = False,
+        quote_identifiers: bool = True,
+        **kwargs: t.Any,
+    ) -> t.Tuple:
+        ...
+
+    def execute_and_fetch(
```

Comment:
move this into fetchall, and then have fetch_one just return 1

Reply:
from fetchall

Reply:
So fetchone should call fetchall and return the first entry?

Reply:
yes, and check the behavior on how it should raise

Reply:
actually i change my mind

Reply:
can't you just do 

with self.transaction():
   self.execute()
   return self.cursor.fetchone()